src/simulation.cpp

```diff
@@ -68,15 +279,99 @@ int thread_id;  //!< ID of a given thread
 // Non-member functions
 //==============================================================================
 
-void openmc_simulation_init_c()
+void initialize_batch()
 {
-  // Determine how much work each process should do
-  calculate_work();
+  // Increment current batch
+  ++simulation::current_batch;
 
-  // Allocate array for matching filter bins
-  #pragma omp parallel
-  {
-    filter_matches.resize(n_filters);
+  if (settings::run_mode == RUN_MODE_FIXEDSOURCE) {
+    int b = simulation::current_batch;
+    write_message("Simulating batch " + std::to_string(b), 6);
+  }
+
+  // Reset total starting particle weight used for normalizing tallies
+  total_weight = 0.0;
+
+  if ((settings::n_inactive > 0 && simulation::current_batch == 1) ||
+      (settings::restart_run && simulation::restart_batch < settings::n_inactive &&
+        simulation::current_batch == simulation::restart_batch + 1)) {
+    // Turn on inactive timer
+    time_inactive.start();
+  } else if ((simulation::current_batch == settings::n_inactive + 1) ||
+   (settings::restart_run && simulation::restart_batch > settings::n_inactive &&
+    simulation::current_batch == simulation::restart_batch + 1)) {
+    // Switch from inactive batch timer to active batch timer
+    time_inactive.stop();
+    time_active.start();
+
+    for (int i = 1; i <= n_tallies; ++i) {
+      // TODO: change one-based index
+      openmc_tally_set_active(i, true);
+    }
+  }
```

Comment:
The conditions on this `if` block are really hard to read.  I would recommend breaking out that logic with something like this:

```c++
  // Determine if this batch is the first inactive or active batch.
  bool first_inactive = false;
  bool first_active = false;
  if (!settings::restart_run) {
    first_inactive = settings::n_inactive > 0 && simulation::current_batch == 1;
    first_active = simulation::current_batch == settings::n_inactive + 1;
  } else if (simulation::current_batch == simulation::restart_batch + 1){
    first_inactive = simulation::restart_batch < settings::n_inactive;
    first_active = simulation::restart_batch > settings::n_inactive;
  }

  // Manage active/inactive timers and activate tallies if necessary.
  if (first_inactive) {
    time_inactive.start();
  } else if (first_active) {
    time_inactive.stop();
    time_active.start();
    for (int i = 1; i <= n_tallies; ++i) {
      // TODO: change one-based index
      openmc_tally_set_active(i, true);
    }
  }
```

When I wrote out the code like this, I noticed that there appears to be an inconsistency when `simulation::restart_batch == settings::n_inactive`.  Is that a nonexistent case because of some fudging on the `restart_batch` variable?  Should it instead be the case, `first_active = !first_inactive;` if we're in the restart run branch of my first proposed `if` block?

Reply:
Hmm, yeah I agree with you. On that branch, we should have that `first_active = !first_inactive`. I'll update this. Thanks for the suggestion!

Reply:
Next time, you can make a [suggested code change](https://blog.github.com/changelog/2018-10-16-suggested-changes/) directly ;)

Reply:
I've only figured out how to make it work with single-line changes.  It looks like the suggestion diff only ever removes one line.  I tried sticking a diff-formated statement in, too, but that just tried to include the actual `+` and `-` characters in the code.  Let me know if you figure it out!